services/src/main/java/org/jboss/windup/web/services/model/RegisteredApplication.java

```diff
@@ -40,6 +42,7 @@
     public static final String REGISTERED_APPLICATION_ID = "registered_application_id";
 
     @Id
+    @Access(AccessType.PROPERTY) // Allow accessing ID without Lazy-loading the entity.
```

Comment:
If you have a relation to this, the ID is in the source entity table. And the proxy returns that id. So Hibernate doesn't need to load anything, it's just there.